akka-docs/rst/scala/http/introduction.rst

```diff
@@ -10,23 +10,105 @@ You get to pick the API level of abstraction that is most suitable for your appl
+this case shown by two separate routes where one returns an ``Item`` that is marshalled into JSON and one accepts
+JSON that is unmarshalled into an ``Order``.
+
+The logic for the marshalling and unmarshalling is provided by the "spray-json" library (Details how to use that here
+:ref:`akka-http-spray-json`) as two methods creating reflection based implementations:
+
+.. includecode:: ../code/docs/http/scaladsl/SprayJsonExampleSpec.scala
+   :include: example
+
+One of the strengths of akka-http is that streaming data is at its heart meaning that both request and response bodies
+can be streamed through the server achieving constant memory usage even for very large requests or responses. Streaming
+responses will be backpressured by the remote client so that the server will not push data faster than the client can
+handle, streaming requests means that the server decides how fast the remote client can push the data of the request
+body.
+
+Example that streams all positive numbers (given infinite time):
+
+.. includecode2:: ../code/docs/http/scaladsl/HttpServerExampleSpec.scala
+   :snippet: all-positive-numbers
+
+
+
+Read more about the details of the high level APIs in the section :ref:`http-high-level-server-side-api`.
+
+Low-level HTTP server APIs
+--------------------------
+The low-level Akka HTTP server APIs allows for handling connections or individual requests by accepting
+``HttpRequest`` s and answering them by producing ``HttpResponse`` s. This is provided by the ``akka-http-core`` module.
+APIs for handling such request-responses as function calls and as a ``Flow[HttpRequest, HttpResponse, _]`` are available.
+
+.. includecode2:: ../code/docs/http/scaladsl/HttpServerExampleSpec.scala
+   :snippet: full-server-example
+
+Read more details about the low level APIs in the section :ref:`http-low-level-server-side-api`.
+
+
+HTTP client API
+---------------
+The client APIs provide methods for calling a HTTP server using the same ``HttpRequest`` and ``HttpResponse`` abstractions
+that Akka HTTP server uses but adds the concept of connection pools to allow multiple requests to the same server to be
+handled more performantly by re-using TCP connections to the server.
+
+Example simple request:
+
+.. includecode:: ../code/docs/http/scaladsl/HttpClientExampleSpec.scala
+   :include: single-request-example
+
+
+Read more about the details of the client APIs in the section :ref:`http-client-side`.
+
+
+
+The modules that make up Akka HTTP
+----------------------------------
 Akka HTTP is structured into several modules:
 
 akka-http-core
   A complete, mostly low-level, server- and client-side implementation of HTTP (incl. WebSockets)
+  Details can be found in sections :ref:`http-low-level-server-side-api` and :ref:`http-client-side`
 
 akka-http
   Higher-level functionality, like (un)marshalling, (de)compression as well as a powerful DSL
```

Comment:
We could also say `(recommended)` here perhaps, as that's the one we recommend for writing servers?